server-common/src/main/java/org/apache/kafka/server/group/share/DefaultStatePersister.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.server.group.share;
+
+import org.apache.kafka.common.annotation.InterfaceStability;
+
+@InterfaceStability.Evolving
+public class DefaultStatePersister {
+  /**
+   * The InitializeShareGroupState API is used by the group coordinator to initialize the share-partition state.
+   * This is an inter-broker RPC authorized as a cluster action.
+   *
+   * @param request InitializeShareGroupStateRequestDTO
+   * @return InitializeShareGroupStateResponseDTO
+   */
+  public InitializeShareGroupStateResponseDTO initializeShareGroupState(InitializeShareGroupStateRequestDTO request) {
```

Comment:
I think all of these methods should be async, so I think the return type of `CompletableFuture<InitializeShareGroupStateResponseDTO>` would be appropriate.

Reply:
Yeah, this was initial attempt - since not callers have been added yet. Was planning on adding that in future PRs. Will add new commit for the same.